modules/globebrowsing/src/tileprovider/tileproviderbyindex.cpp

```diff
@@ -139,7 +171,16 @@ int TileProviderByIndex::minLevel() {
 }
 
 int TileProviderByIndex::maxLevel() {
-    return _defaultTileProvider->maxLevel();
+    int result = _defaultTileProvider->maxLevel();
+
+    using K = TileIndex::TileHashKey;
+    using V = std::unique_ptr<TileProvider>;
+    for (std::pair<const K, V>& it : _providers) {
+        TileIndex index = TileIndex(it.first);
+        result = std::max(result, static_cast<int>(index.level));
```

Comment:
Does this set the max level to the maximum of all the tile providers? Won't that mean that the other tile providers will go out of range?

Reply:
Yes, but that is fine and already supported by the GlobeBrowsing subsystem.   If a layer is asked to generate a tile that is higher resolution that it is capable of, it will reuse the last possible tile and create smaller and smaller views on it by manipulating the uv coordinates